kubernetes/migration/spec.md

```diff
@@ -0,0 +1,33 @@
+# K8s/CoreDNS Corefile Migration Tools
+
+This Go library provides a set of functions to help handle migrations of CoreDNS Corefiles to be compatible with new versions of CoreDNS.  The task of upgrading CoreDNS is the responsibility of a variety of Kubernetes management tools (e.g. kubeadm and others), and the precise behavior may be different for each one.  This library abstracts some basic helper functions that make this easier to implement.
+
+## Proposed functions:
+
+`Deprecated(fromVersion, toVersion, corefile string) []string`: returns a list of deprecated plugins or directives present in the Corefile.
```

Comment:
is `version` of CoreDNS or of Kubernetes ?

Reply:
* we just return a set of keyword, mixing plugin names and options in the plugin ?
* what if the syntax of an option change ? (or should we say that cannot happen ?)

Reply:
> is `version` of CoreDNS or of Kubernetes ?

CoreDNS

Reply:
Its a list of warnings/ messages like:

"plugin: 'proxy' is deprecated", "directive 'upstream' in 'kubernetes' is deprecated".

From the perspective of something using the tool, these would just be displayed to a user... There isn't much the receiving end can act on even if we try to make these complex structs.